epersongroups.md

```diff
@@ -0,0 +1,390 @@
+
+### Get Sub Groups in a single parent EPerson Group
+
+**GET /api/eperson/groups/<:uuid>/subgroups**
+
+Retrieves direct child groups of the given group.
+Requesting indirect child groups is not supported
+
+Sample:
+* GET /rest/api/eperson/groups/3b1de75d-5bf9-4ca3-bf4d-42bc8abd0d42/subgroups
+
+```json
+{
+    "_embedded": {
+      "groups": [
+        {
+          "id": "617cf46b-535c-42d5-9d22-327ce2eff6dc",
+          "uuid": "617cf46b-535c-42d5-9d22-327ce2eff6dc",
+          "name": "Administrator",
+          "handle": null,
+          "metadata": {
+              "dc.title": [
+                {
+                  "value": "Administrator",
+                  "language": null,
+                  "authority": "",
+                  "confidence": -1,
+                  "place": 0
+                }
+              ]
+          },
+          "permanent": true,
+          "type": "group",
+          "_links": {
+            "subgroups": {
+              "href": "https://dspace7.4science.it/dspace-spring-rest/api/eperson/groups/617cf46b-535c-42d5-9d22-327ce2eff6dc/subgroups"
+            },
+            "parentgroups": {
+              "href": "https://dspace7.4science.it/dspace-spring-rest/api/eperson/groups/617cf46b-535c-42d5-9d22-327ce2eff6dc/parentgroups"
+            },
+            "epersons": {
+              "href": "https://dspace7.4science.it/dspace-spring-rest/api/eperson/groups/617cf46b-535c-42d5-9d22-327ce2eff6dc/epersons"
+            },
+            "self": {
+              "href": "https://dspace7.4science.it/dspace-spring-rest/api/eperson/groups/617cf46b-535c-42d5-9d22-327ce2eff6dc"
+            }
+          }
+        }
+      ]
+    },
+    "_links": {
+      "self": {
+        "href": "https://dspace7.4science.it/dspace-spring-rest/api/eperson/groups/3b1de75d-5bf9-4ca3-bf4d-42bc8abd0d42/subgroups"
+      }
+    },
+    "page": {
+      "number": 0,
+      "size": 10,
+      "totalPages": 1,
+      "totalElements": 10
+    }
+}
+```
+
+### Add a Group to a parent Group
+
+**POST /api/eperson/groups/<:parentgroupuuid>/subgroups**
+
+To add a child group to a parent group, perform a POST to the Sub groups of a Group endpoint when logged in as admin.
+
+The actual sub group is part of the body using the uri-list
+
+Example:
+```bash
+curl -i -X POST "https://dspace7.4science.it/dspace-spring-rest/api/eperson/groups/617cf46b-535c-42d5-9d22-327ce2eff6dc/subgroups"
+  -H "Content-Type:text/uri-list"
+  -d "https://dspace7.4science.it/dspace-spring-rest/api/eperson/groups/05e3dbb8-332b-4487-a3f9-d78431b6cc02"
+```
+
+The group is mandatory
+
+Return codes:
+* 204: if the update succeeded 
+* 401 Forbidden - if you are not authenticated
+* 403 Unauthorized - if you are not logged in with sufficient permissions 
+* 404 Not found - if the parent group doesn't exist
+* 409 Conflict - if adding the group would create a cyclic reference (ie. A -> B -> C, -> mean contains we cannot add A to C)
```

Comment:
I'm not sure if we can use the 409 for anything different than versioning as this is the typical scenario where 409 is used. Maybe we should just use the 422 and think about adding later more details in the response body

Reply:
Yes, I think this could just be changed to a 422.  It looks (to me) like 409 is supposed to be for PUT requests that update a resource in an incorrect manner.

Reply:
I've modified this to use 422